programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -106,43 +115,31 @@ pub mod gateway {
         Ok(())
     }
 
+    // deposit SOL into this program and the `receiver` on ZetaChain zEVM
+    // will get corresponding ZRC20 credit. The `receiver` should be a contract
+    // on zEVM and the `message` will be used as input data for the contract call.
+    // The `receiver` contract on zEVM will get the SOL ZRC20 credit and receive the `message`.
     pub fn deposit_and_call(
         ctx: Context<Deposit>,
         amount: u64,
         receiver: [u8; 20],
         message: Vec<u8>,
     ) -> Result<()> {
         require!(message.len() <= 512, Errors::MemoLengthExceeded);
```

Comment:
Should we check also for message.len too short?

Reply:
0 size message seems to be no problem; the app contract can expect empty message; maybe there is use case for that. Let's not be too pedantic here.